File: 1两数之和/src/Solution.java
import java.util.HashMap;
import java.util.Map;

public class Solution {
    public int[] twoSum(int[] nums, int target) {
        //用hash表加快查找
        Map<Integer, Integer> hashMap = new HashMap<>();
        for(int i=0; i< nums.length; i++){
            if(hashMap.containsKey(target-nums[i])){
                return new int[]{i, hashMap.get(target - nums[i])};
            }else {
                hashMap.put(nums[i], i);
            }
        }
        return null;
    }
}

File: 1两数之和/src/Solution.cpp
//
// Created by Frank on 2025/10/16.
//
#include <vector>
#include <map>
using namespace std;

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int, int> hashMap;
        vector<int> ans;
        for (int i=0; i< nums.size(); i++) {
            if (hashMap.find(target - nums[i]) != hashMap.end()) {
                ans.push_back(hashMap.at(target - nums[i]));
                ans.push_back(i);
                break;
            } else {
                hashMap[nums[i]] = i;
            }
        }

        return ans;
    }
};